core-bundle/src/Resources/public/tips.js

```diff
@@ -0,0 +1,127 @@
+            }
+
+            if (!text) {
+                return;
+            }
+
+            clearTimeout(timer);
+            tip.style.willChange = 'display,contents';
+
+            timer = setTimeout(function () {
+                const position = el.getBoundingClientRect();
+                tip.innerHTML = `<div>${text}</div>`;
+                tip.style.top = `${(window.scrollY + position.top + y)}px`;
+                tip.style.left = `${(window.scrollX + position.left + x)}px`;
+                tip.style.display = 'block';
+                tip.style.willChange = 'auto';
+                tip.setAttribute('aria-hidden', false);
+
+                if (!tip.parentNode && document.body) {
+                    document.body.append(tip);
+                }
+            }, 1000)
+        })
+        el.addEventListener('mouseleave', function () {
+            if (!useContent && text && !el.hasAttribute('title')) {
+                el.setAttribute('title', text);
+            }
+
+            clearTimeout(timer)
+            tip.style.willChange = 'auto';
+
+            if (tip.style.display === 'block') {
+                tip.style.willChange = 'display';
+                timer = setTimeout(function () {
+                    tip.style.display = 'none';
+                    tip.style.willChange = 'auto';
+                    tip.setAttribute('aria-hidden', true);
+                }, 100)
+            }
+        })
+
+        // Hide tooltip when clicking a button (usually an operation icon in a wizard widget)
+        const action = el.closest('button, a');
+        if (action) {
+            action.addEventListener('click', function () {
+                clearTimeout(timer);
+                tip.style.display = 'none';
+                tip.style.willChange = 'auto';
+                tip.setAttribute('aria-hidden', true);
+            })
+        }
+    }
+
+    function setup(node) {
+        node.querySelectorAll('p.tl_tip').forEach(function (el) {
+            init(el, 0, 23, true);
+        });
+
+        node.querySelectorAll('#home').forEach(function (el) {
+            init(el, 6, 42);
+        });
+
+        node.querySelectorAll('#tmenu a[title]').forEach(function (el) {
+            init(el, 0, 42);
+        });
+
+        node.querySelectorAll('a[title][class^="group-"]').forEach(function (el) {
+            init(el, -6, 27);
+        });
+
+        node.querySelectorAll('a[title].navigation').forEach(function (el) {
+            init(el, 25, 32);
+        });
+
+        node.querySelectorAll('img[title]').forEach(function (el) {
+            init(el, -9, el.classList.contains('gimage') ? 60 : 30);
+        });
+
+        ['a[title]', 'input[title]', 'button[title]', 'time[title]', 'span[title]'].forEach(function(selector) {
+            node.querySelectorAll(selector).forEach(function (el) {
+                init(el, -9, ((selector === 'time[title]' || selector === 'span[title]') ? 26 : 30));
+            });
+        });
+    }
+
```

Comment:
What you could do is collect those as an array, then dispatch an event for other devs to add their own elements, and then init the array?

Reply:
yeah there are many options. I'm just unsure whether we want to add an API for it since any API we add would then need to be supported. If we add an API that noone uses then we add ourselves some unnecessary overhead. We could still add an API on request 🤷 
On the other hand, once we have "real" JS modules (when I rewrite it again) we can export the necessary method so anyone can use that in their own code.